src/frequenz/sdk/data_ingestion/resampling/timeseries_resampler.py

```diff
@@ -0,0 +1,118 @@
+"""
+TimeSeriesResampler class that delegates resampling to individual resmaplers.
+
+Copyright
+Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+License
+MIT
+"""
+import logging
+from datetime import datetime
+from typing import Callable, Dict, Optional, Sequence
+
+import pytz
+
+from frequenz.sdk.core.timeseries import Sample, TimeSeriesId
+from frequenz.sdk.data_ingestion.resampling.individual_timeseries_resampler import (
+    IndividualTimeSeriesResampler,
+)
+
+logger = logging.Logger(__name__)
+
+ResamplingFunction = Callable[[Sequence[Sample], float], float]
+
+DEFAULT_BUFFER_SIZE = 100
+
+
+class TimeSeriesResampler:
+    """TimeSeriesResampler class that delegates resampling to individual resmaplers."""
+
+    def __init__(
+        self,
+        period_s: float,
+        max_data_age_in_periods: float,
+        resampling_function: ResamplingFunction,
+    ) -> None:
+        """Initialize the TimeSeriesResampler.
+
+        Args:
+            period_s: resampling frequency, in seconds
+            max_data_age_in_periods: max age that samples shouldn't exceed in order
+                to be used in the resampling function
+            resampling_function: function to be applied to a sequence of samples within
+                a resampling period to produce a single output sample
+        """
+        self._period_s = period_s
+        self._max_data_age_in_periods: float = max_data_age_in_periods
+        self._resampling_function: ResamplingFunction = resampling_function
+        self._resamplers: Dict[TimeSeriesId, IndividualTimeSeriesResampler] = {}
+
+    def add_time_series(self, time_series_id: TimeSeriesId) -> None:
+        """Create a new resampler for a specific time series.
+
+        Args:
+            time_series_id: time series id
+
+        Raises:
+            ValueError: if resampler for the provided timer_series_id already exists
+        """
+        resampler = self._resamplers.get(time_series_id)
+        if resampler is None:
+            raise ValueError(
+                f"Resampler for time series {time_series_id} already exists!"
+            )
+
+        self._resamplers[time_series_id] = IndividualTimeSeriesResampler(
+            buffer_size=DEFAULT_BUFFER_SIZE,
+            period_s=self._period_s,
+            max_data_age_in_periods=self._max_data_age_in_periods,
+            resampling_function=self._resampling_function,
+        )
+
+    def remove_timeseries(self, time_series_id: TimeSeriesId) -> None:
+        """Remove a resampler for a specific time series.
+
+        Args:
+            time_series_id: time series id, for which to remove the resampler
+
+        Raises:
+            ValueError: if resampler for the provided timer_series_id doesn't exist
+        """
+        resampler = self._resamplers.get(time_series_id)
+        if resampler is None:
+            raise ValueError(f"No resampler for time series {time_series_id} found!")
```

Comment:
We can also silence it when there's no resampler for a specific `time_series_id`, but `del self._resamplers[time_series_id]` would raise a `KeyError` if `time_series_id` wouldn't be in the dict so I just added a bit more descriptive error message here (perhaps `ValueError` could be changed to `KeyError`).

Reply:
If you keep it, I agree it should be a `KeyError`, and also it would be better to do a try/except and raise the new error inside the except clause so they get properly chained.

I'm fine with both approaches, both have pros and cons.

Reply:
`RingBuffer` has been removed.